backend/ethereum/channel/conclude.go

```diff
@@ -35,31 +36,40 @@ const secondaryWaitBlocks = 2
 //   - if found, channel is already concluded and success is returned
 //   - if none found, conclude/concludeFinal is called on the adjudicator
 // - it waits for a Concluded event from the blockchain.
+// nolint: funlen
 func (a *Adjudicator) ensureConcluded(ctx context.Context, req channel.AdjudicatorReq, subStates channel.StateMap) error {
 	// Listen for Concluded event.
-	watchOpts, err := a.NewWatchOpts(ctx)
+	events := make(chan *subscription.Event, 10)
+	subErr := make(chan error, 1)
+	sub, err := subscription.NewEventSub(ctx, a.ContractBackend, a.bound, updateEventType(req.Params.ID()), startBlockOffset)
 	if err != nil {
-		return errors.WithMessage(err, "creating watchOpts")
+		return errors.WithMessage(err, "subscribing")
 	}
-	events := make(chan *adjudicator.AdjudicatorChannelUpdate)
-	sub, err := a.contract.WatchChannelUpdate(watchOpts, events, [][32]byte{req.Params.ID()})
-	if err != nil {
-		err = cherrors.CheckIsChainNotReachableError(err)
-		return errors.WithMessage(err, "creating subscription failed")
+	defer sub.Close()
+	// Check for past event
+	if err := sub.ReadPast(ctx, events); err != nil {
```

Comment:
Because i did not want to change the behaviour of the function.  
The function read the past events first, aborted if any, and then continued.  
This is different from just reading all events.  

It would surely be better if we could simplify all these functions, but i did not want to do that in this MR since it was out of scope.

Reply:
We should focus on having a correct and clean implementation first, and then worry about how much it is consistent with the prior implementation.

The idea was that we can plug in our new subscription in most places without having to merge the past and future event streams manually. Are you saying that this is not possible?

Reply:
It is possible, but i am not comfortable with just changing the protocol i so many places without concrete concept.  
The past and future event stream merging is done here on purpose, since they are treated *differently*,

Reply:
OK, we can keep as is for now.